gateway/src/resty/resolver/http.lua

```diff
@@ -8,9 +8,23 @@ local _M = setmetatable({}, { __index = resty_http })
 
 local mt = { __index = _M }
 
-function _M.new()
+function _M.new(opts)
+  opts = opts or { }
   local http = resty_http:new()
 
+  local timeouts = opts.timeouts
+  if timeouts then
+    ngx.log(ngx.DEBUG, 'setting timeouts (secs), connect_timeout: ', timeouts.connect_timeout,
+      ' send_timeout: ', timeouts.send_timeout, ' read_timeout: ', timeouts.read_timeout)
+    -- lua-resty-http uses nginx API for lua sockets
+    -- in milliseconds
+    -- https://github.com/openresty/lua-nginx-module?tab=readme-ov-file#tcpsocksettimeouts
+    local connect_timeout = timeouts.connect_timeout and timeouts.connect_timeout * 1000
+    local send_timeout = timeouts.send_timeout and timeouts.send_timeout * 1000
+    local read_timeout = timeouts.read_timeout and timeouts.read_timeout * 1000
+    http:set_timeouts(connect_timeout, send_timeout, read_timeout)
```

Comment:
Can we handle this inside `gateway/src/resty/http/proxy.lua` connect() method instead? This `new()` return a lua-resty-http instance, so you should be able to call `set_timeouts()` there.